ros_buildfarm/status_page.py

```diff
@@ -873,6 +928,98 @@ def _get_blocked_releases_info(config_url, rosdistro_name, repo_names=None):
     return repos_info
 
 
+def _has_source_entry(repo_name, dist_file):
+    return repo_name in dist_file.repositories and \
+        dist_file.repositories[repo_name].source_repository is not None and \
+        dist_file.repositories[repo_name].source_repository.url is not None
+
+
+def _prev_rosdistro(index, rosdistro_name):
+    """Given current rosdistro name, return the previous."""
+    valid_rosdistro_names = list(index.distributions.keys())
+    valid_rosdistro_names.sort()
+
+    # skip distributions with a different type if the information is available
+    distro_type = index.distributions[rosdistro_name].get('distribution_type')
+    if distro_type is not None:
+        valid_rosdistro_names = [
+            n for n in valid_rosdistro_names
+            if distro_type == index.distributions[n].get('distribution_type')]
+
+    try:
+        i = valid_rosdistro_names.index(rosdistro_name)
+    except ValueError:
+        raise ValueError('Distribution key not found in list of valid distributions.')
+
+    if i == 0:
+        raise ValueError('No previous distribution found.')
+
+    return valid_rosdistro_names[i - 1]
+
+
+def _rosdistro_or_latest(index, rosdistro_name=None):
+    """Return rosdistro_name if specified, else return lastest distro."""
+    if rosdistro_name is None:
+        valid_rosdistro_names = list(index.distributions.keys())
+        valid_rosdistro_names.sort()
+        rosdistro_name = valid_rosdistro_names[-1]
+    return rosdistro_name
+
+
+def _released_repos(distro):
+    for repo_name in distro.repositories.keys():
+        if _is_released(repo_name, distro):
+            yield repo_name
+
+
+def _source_entry_repos(distro):
+    for repo_name in distro.repositories.keys():
+        if _has_source_entry(repo_name, distro):
+            yield repo_name
+
+
+def _package_dependencies(walker, package):
+    try:
+        for dep in walker.get_recursive_depends(
+                package, ['build', 'buildtool', 'run', 'test'], ros_packages_only=True,
+                limit_depth=1):
+            yield dep
+    except AssertionError as e:
+        print(e, file=sys.stderr)
+
+
+def _released_packages(distro, repo_names=None):
```

Comment:
I don't see where this function is ever used with a `repo_names` being `None`?

Reply:
Made `repo_names` required argument in 08a74fba9a428b7528cb67816e246e7c3bc15598